packages/astro/src/core/render-context.ts

```diff
@@ -519,16 +522,24 @@ export class RenderContext {
 		};
 	}
 
-	clientAddress() {
-		const { pipeline, request } = this;
-		if (clientAddressSymbol in request) {
-			return Reflect.get(request, clientAddressSymbol) as string;
-		}
+	getClientAddress() {
+		const { pipeline, request, routeData, clientAddress } = this;
 
-		if (request.body === null) {
+		if(routeData.prerender) {
 			throw new AstroError(AstroErrorData.PrerenderClientAddressNotAvailable);
 		}
 
+		if(clientAddress) {
+			return clientAddress;
+		}
+
+		// Legacy, should not need to get here.
```

Comment:
Can the word TODO be added here so that we can find it back easily whenever we'll do Astro 6 or later?